### ISSUE TITLE
Ignore folders in mail source

### ISSUE THREAD

**Issue by krillin666** [seanbreckenridge/HPI#39]
Hey @seanbreckenridge how are you doing ?
I’ve noticed that my promnesia indexing is taking longer than usual, and found out that it might be because it is indexing all my **Trash** folders from my different emails. Is it possible to exclude certain folder of indexing using the Thunderbird method ?

**Comment by seanbreckenridge** [comment]
Ah yeah, for the `imap` portion I have this in my config:

```python
# this is all custom, and may not work for you -- this
# is how I filter certain Junk/Trash paths from my mail

FILTER_PARTS: Set[str] = {".notmuch", "Trash", "Spam", "Junk"}


def filter_mail_path(p: Path) -> bool:
    pset = set(p.parts[1:])
    if any(f in pset for f in FILTER_PARTS):
        return False
    return True


class mail:
    # locally synced IMAP mailboxes using mbsync
    class imap:
        # path[s]/glob to the the mailboxes/IMAP files
        # you could also just do something like:
        # mailboxes = "~/Documents/mail/*@*"
        # to match any files in that directory with '@' in them
        #
        # to confirm this is matching your files, can do:
        # hpi query my.mail.imap.files -s
        mailboxes = MAILDIR

        # filter function which filters the input paths
        filter_path: Optional[Callable[[Path], bool]] = filter_mail_path

```

but since the mbox just loads the one huge file, you cant exclude extensions or paths as far as I understand.

Could exclude the trash in the export in the first place (though not sure exactly how to do that)

Will do an export and try to debug to see if the path is accessible in some way

**Comment by seanbreckenridge** [comment]
Hmm, trying to reproduce what you might run into, do you remember how you setup the export?

I selected my inbox like this in Thunderbird:

![image](https://user-images.githubusercontent.com/7804791/187370105-449773b6-d0dc-4cdf-a3b6-5d2881ff1cd6.png)

And then followed what I described here [in the mail doc](https://github.com/seanbreckenridge/HPI/blob/master/doc/MAIL_SETUP.md#mymailimap):

`Tools > ImportExportToolsNg > Export all messages in the Folder > Plain Text Format`

That way is just my inbox and not trash/spam that are getting included. Are you doing something else?

**Comment by krillin666** [comment]
> Hmm, trying to reproduce what you might run into, do you remember how you setup the export?
> 
> I selected my inbox like this in Thunderbird:
> 
> ![image](https://user-images.githubusercontent.com/7804791/187370105-449773b6-d0dc-4cdf-a3b6-5d2881ff1cd6.png)
> 
> And then followed what I described here [in the mail doc](https://github.com/seanbreckenridge/HPI/blob/master/doc/MAIL_SETUP.md?rgh-link-date=2022-08-30T06%3A57%3A34Z#mymailimap):
> 
> `Tools > ImportExportToolsNg > Export all messages in the Folder > Plain Text Format`
> 
> Edit: Or just right clicking and hitting `Export remote folder`, which includes all ~10000 messages in my account, not just the 2000 local cache.
> 
> That way is just my inbox and not trash/spam that are getting included. Are you doing something else, possibly because of the scheduling?

Sorry for the late reply, I'm using the automatic scheduling feature from the addon. And thus I cannot see how to exclude certain folders. I'll try and see if your filter works !

**Comment by krillin666** [comment]
Hum, it is still indexing the *deleted, trash, etc* folders. Though I'm using `mbox` instead of IMAP

**Comment by seanbreckenridge** [comment]
Filtering is only a thing for `imap` because i have the files laid out in my filesystem, its not all in a single giant `mbox` file

As far as I remember, the thunderbird addon exports the whole account into `mbox`, and last time I debugged I couldnt find anything like a 'folder' or 'path' in the metadata that described where the email was from. Will try doing an automatic export with thunderbird to see if I can find anything like that

**Comment by seanbreckenridge** [comment]
Just leaving this here incase I want to try and reproduce this later, setup daily backups and going to leave thunderbird open so it'll do a backup sometimes in the next day (could trigger manually but want to make sure I'm not doing something custom there)

![image](https://user-images.githubusercontent.com/7804791/223932836-b0f829fe-b3ff-4232-ba4a-4c32ee129076.png)
![image](https://user-images.githubusercontent.com/7804791/223932847-04c4155d-9085-4902-96b0-609bfac0bb97.png)

**Comment by krillin666** [comment]
Yeah I was already using the `just mail files` though I don't see any option for the scheduling to exclude *Trahs/Deleted* folders

**Comment by seanbreckenridge** [comment]
sorry this takes a while, I can only really do a test once per day since I don't want to do something custom and I want it to run with scheduling. I had it exporting to msf files before, I changed this option to export to mbox (I think? will test tomorrow):
![image](https://user-images.githubusercontent.com/7804791/225134015-0d508ca5-e1c3-4974-bbe1-83ec60ece9a4.png)
